### PR TITLE
Create CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by lkellogg**
Github recommends that projects meet a variety of community standards (https://github.com/geodynamics/aspect/community). Aspect currently meets most of these. This pull request adds a community code of conduct to Aspect to meet an additional one of these standards. I have very lightly modified the template provided by Github.

### CODE REVIEW

**Comment by gassmoeller** [comment]
:+1: I like this!

**Comment by ljhwang** [comment]
The code of conduct should reflect that this project is part of the CIG community that has its own code of conduct.
As such, we should consider how this is enforced.  Should the maintainers also be the enforcers or as part of the CIG community, should CIG leadership review such cases?  Either way, the email address help@geodynamics is not an appropriate address.
Enforcement should be consistent with Our Responsibilities.
Confidentiality cannot be promised under Title IX

**Comment by gassmoeller** [comment]
So @ljhwang do you require changes before merge? Then I remove the ready to merge label.

**Comment by gassmoeller** [comment]
Ok, we should address the changes before merging this.

**Comment by ljhwang** [comment]
A new email address has been set up in lieu of help@: private@geodynamics.org.  This will forward explicitly (no alias) to the Director, Assoc. Dir, Chair EC, and Chair SSC. Are these the correct people?

**Comment by gassmoeller** [comment]
I took the liberty to update this PR with Lorraine's suggestions. @ljhwang: could you take a look if I included everything correctly? If you have suggestions feel free to provide them. If we finish this we can also include it as an example code of conduct into our manual template (or as a separate document).

**Comment by ljhwang** [comment]
Thanks Rene.
@lkellogg, any further comments?
Looks good to me. I think it would be good to push his out to include both in the template and the upcoming workshop.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to 
+making participation in our project and our community a harassment-free experience for everyone, 
+regardless of age, body size, disability, ethnicity, gender identity and expression, technology choices and 
+level of experience, nationality, personal appearance, race, religion (or lack thereof), or sexual identity and orientation.
```
Comment (bangerth) [inline CODE_OF_CONDUCT.md]: These are all ordered alphabetically, except for "technology choices and level or experience".

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+* Focusing on what is best for the community
+* Showing empathy towards other community members
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery and unwelcome sexual attention or advances
+* Trolling, insulting/derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or electronic address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a professional setting
```
Comment (bangerth) [inline CODE_OF_CONDUCT.md]: "Other conduct" is not wrong, but it's not an "*Example* of unacceptable behavior"

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+## Our Responsibilities
+
+Project maintainers are responsible for clarifying the standards of acceptable behavior and are expected to take appropriate and fair corrective action in response to any instances of unacceptable behavior.
+
+Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, or to ban temporarily or permanently any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful.
```
Comment (bangerth) [inline CODE_OF_CONDUCT.md]: Can we add a sentence like "All of these things are last resorts. Our goal is to have a friendly and inclusive community where everyone feels welcome to participate and express themselves, but conduct by individuals that jeopardizes the harmony of the project will need to be addressed."

Not sure this is the right wording, but you get the idea.

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting any [maintainer](README.md#more-information) of the project, or CIG leadership at <private@geodynamics.org>. CIG leadership will review and investigate all complaints, and will respond in a way that it deems appropriate to the circumstances. CIG leadership is obligated to maintain confidentiality with regard to the reporter of an incident, although confidentiality cannot be promised in all situations under the Title IX law of the United States. Further details of specific enforcement policies may be posted separately.
+
+Project maintainers who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent repercussions as determined by other members of the project's leadership.
+
+## Attribution
+
+This Code of Conduct is adapted from the [Contributor Covenant][homepage], version 1.4, available at [http://contributor-covenant.org/version/1/4][version] and draws on Community Code of Conduct of the Computational Infrastructure for Geodynamics, available at [https://geodynamics.org/cig/about/code-conduct/].
```
Comment (tjhei) [inline CODE_OF_CONDUCT.md]: the link at the end doesn't show up as intended, I think. Remove brackets?

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+[homepage]: http://contributor-covenant.org
+[version]: http://contributor-covenant.org/version/1/4/
```
Comment (tjhei) [inline CODE_OF_CONDUCT.md]: what does this do?

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (tjhei) [reply CODE_OF_CONDUCT.md]: maybe just remove "Examples of" from the caption?

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (tjhei) [reply CODE_OF_CONDUCT.md]: And why is that written with an "and"? Are you trying to say "level of experience" applying to "technology"?

**Comment by gassmoeller** [comment]
I think I addressed all comments. Is this ready?

**Pull request merged**